netcdf4/src/main/java/ucar/nc2/jni/netcdf/Nc4Iosp.java

```diff
@@ -2339,8 +2350,10 @@ private void createGroup(Group4 g4) throws IOException {
 
     // dimensions
     for (Dimension dim : g4.g.getDimensions()) {
-      int dimid = addDimension(g4.grpid, dim.getShortName(), dim.getLength());
+      int dimLength = dim.isUnlimited() ? Nc4prototypes.NC_UNLIMITED : dim.getLength();
+      int dimid = addDimension(g4.grpid, dim.getShortName(), dimLength);
```

Comment:
This is the relevant bit. In order to create an unlimited dimension, we must pass a `dimLength` of `NC_UNLIMITED` to `addDimension()`. The value of `NC_UNLIMITED` is `0`. So, your specific example worked ONLY because the length of your unlimited dimension was `0` at the time. For any other value, it would have failed. Maybe you can give it a shot real quick and double-check that I'm right?